### PR TITLE
[#259] Provide device registry AMQP API in hono-service-base.

### PR DESCRIPTION
**Pull request by b-abel**
Introduced base class DeviceRegistryAmqpServerBase and moved logic to there.

Signed-off-by: Abel Buechner <Abel.Buechner@bosch-si.com>

### CODE REVIEW

**Comment by b-abel** [comment]
The intention here is the same as in PR #281: to provide the AMQP server for the Device Registry in hono-service-base. The change there got lost in recent refactorings.

File: service-base/src/main/java/org/eclipse/hono/service/registration/DeviceRegistryAmqpServerBase.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+import org.eclipse.hono.service.amqp.AmqpServiceBase;
+import org.eclipse.hono.service.amqp.Endpoint;
+import org.eclipse.hono.service.auth.AuthenticationConstants;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+
+import java.util.Objects;
+
+/**
+ * Spring component that serves as the default implementation of Hono's device registry.
```
Comment (sophokles73) [inline service-base/src/main/java/org/eclipse/hono/service/registration/DeviceRegistryAmqpServerBase.java]: It doesn't seem to be a Spring `Component` (no annotation)
It also is not the default implementation of Hono's device registry, is it? It is an abstract base class that can be used to create an implementation based on AMQP though ...

Comment (b-abel) [inline service-base/src/main/java/org/eclipse/hono/service/registration/DeviceRegistryAmqpServerBase.java]: You are right. I should have changed that.

**Comment by sophokles73** [comment]
@b-abel can you squash and rebase?

**Comment by b-abel** [comment]
Rebased and squashed.

**Pull request merged**